pyvisa_py/usb.py

```diff
@@ -306,6 +306,55 @@ def list_resources() -> List[str]:
             )
         return out
 
+    def read(self, count: int) -> Tuple[bytes, StatusCode]:
+        """Reads data from device or interface synchronously.
+
+        Corresponds to viRead function of the VISA library.
+
+        Parameters
+        -----------
+        count : int
+            Number of bytes to be read.
+
+        Returns
+        -------
+        bytes
+            Data read from the device
+        StatusCode
+            Return value of the library call.
+
+        """
+
+        def _usb_reader():
+            """Data reader identifying usb timeout exception."""
+            try:
+                return self.interface.read(count)
+            except usb.USBError as exc:
+                if exc.errno in (errno.ETIMEDOUT, -errno.ETIMEDOUT):
+                    raise USBTimeoutException()
+                raise
+
+        supress_end_en, _ = self.get_attribute(ResourceAttribute.suppress_end_enabled)
+
+        if supress_end_en:
+            raise ValueError(
+                "VI_ATTR_SUPPRESS_END_EN == True is currently unsupported by pyvisa-py"
+            )
+
+        term_char, _ = self.get_attribute(ResourceAttribute.termchar)
+        term_char_en, _ = self.get_attribute(ResourceAttribute.termchar_enabled)
+
+        return self._read(
+            _usb_reader,
+            count,
+            lambda current: False,  # USBTMC can return partial message (i.e.,
```

Comment:
Here the end indicator checker does not care about the termination character, only about the fact that the underlying hardware signaled it is done transmitting a message. For GPIB, this is done by checking the state of hardware line, for USB it is always trivially true since by construction a BulkIn message is complete when received. Using this kind of notification is more efficient than checking for term chars.

The ``_read`` implementation takes care of ensuring we read till a term char if we were requested to do so. Actually it is at this layer that support for ``suppress_end`` is implemented so I would appreciate if while removing this implementation, you could also get rid of the error forbidding the use of ``suppress_end``.